src/components/Modal/Votes/VotingModal/VotingModalSurvey.tsx

```diff
@@ -30,58 +39,78 @@ export function VotingModalSurvey({
   onCastVote,
   castingVote,
   proposalPageState,
+  totalVpOnProposal,
+  shouldGiveReason = false,
+  voteWithSurvey,
 }: VotingModalSurveyProps) {
   const [survey, setSurvey] = useState<Survey>([])
   const t = useFormatMessage()
   const { selectedChoice, showVotingError, retryTimer } = proposalPageState
-  const reactionSelected = useMemo(
-    () => survey.some((topicFeedback) => topicFeedback.reaction != Reaction.EMPTY),
-    [survey]
-  )
+
+  const { startTimer, time } = useTimer(REASON_TIMER_SECONDS)
+
+  const [isCastDisabled, setIsCastDisabled] = useState(true)
+
+  const { formatNumber } = useIntl()
+
+  const {
+    formState: { errors },
+    control,
+    watch,
+  } = useForm<Reason>({ defaultValues: { reason: '' }, mode: 'onTouched' })
+
+  useEffect(() => {
+    setIsCastDisabled((shouldGiveReason && (!!errors.reason || time > 0)) || showVotingError)
```

Comment:
Couldn't this be a variable?
`const isCastDisabled = shouldGiveReason && (!!errors.reason || time > 0)) || showVotingError`